src/checks.py

```diff
@@ -312,12 +312,32 @@ def check_git_revision(state: State) -> Optional[str]:
 def check_blacklisted_files(state: State) -> Optional[str]:
     blacklist = [".git", ".gitignore", ".mvn", "mvnw", "mvnw.cmd", "Jenkinsfile"]
     commands = [
-        f"test $(find {state.source_dir} -name {item} | wc -l) -eq 0"
+        f"find {state.source_dir} -name {item} | ifne bash -c 'cat && false'"
         for item in blacklist
     ]
     return _check_sh(commands)
 
 
+@check("No .gitignore-d files in git checkout", hide_if_passing=True)
+def check_gitignore_in_repo(state: State) -> Optional[str]:
+    return _check_sh(
+        "git status --ignored --short | ifne bash -c 'cat && false'",
+        workdir=state.git_dir,
+    )
+
+
+@check("No .gitignore-d files in source release", hide_if_passing=False)
+def check_gitignore_in_release(state: State) -> Optional[str]:
+    sh("git init --quiet && git add .", workdir=state.source_dir)
```

Comment:
Yeah this is a bit wonky, but it's the best I could come up with without having to reimplement the `.gitignore` matching logic. Methinks it's Good Enough (TM)